// react-tab-pane/lib/Tab.js
'use strict';

Object.defineProperty(exports, '__esModule', {
    value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _reactAddons = require('react/addons');

var _reactAddons2 = _interopRequireDefault(_reactAddons);

var _reactVendorPrefix = require('react-vendor-prefix');

var _reactVendorPrefix2 = _interopRequireDefault(_reactVendorPrefix);

var Tab = _reactAddons2['default'].createClass({
    displayName: 'Tab',

    propTypes: {
        id: _reactAddons2['default'].PropTypes.string.isRequired,
        name: _reactAddons2['default'].PropTypes.string.isRequired
        //active: React.PropTypes.boolean
    },

    onClick: function onClick() {
        this.props.selectTab(this);
    },

    render: function render() {
        var classes = ['Tab'];
        if (this.props.active) {
            classes.push('active');
        }
        var tabStyle = undefined;
        if (this.props.orientation === 'vertical') {
            tabStyle = {
                cursor: 'pointer',
                display: 'block'
            };
        } else {
            tabStyle = {
                cursor: 'pointer',
                display: 'inline-block'
            };
        }

        var styles = _reactVendorPrefix2['default'].prefix({ tabStyle: tabStyle });

        return _reactAddons2['default'].createElement(
            'div',
            { className: classes.join(' '), style: styles.tabStyle, onClick: this.onClick },
            this.props.name
        );
    }
});

exports['default'] = Tab;
module.exports = exports['default'];
